drivers/wifi/nrf700x/src/wpa_supp_if.c

```diff
@@ -711,6 +711,13 @@ int nrf_wifi_wpa_supp_authenticate(void *if_priv, struct wpa_driver_auth_params
 	vif_ctx_zep = if_priv;
 	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
 
+#ifdef CONFIG_NRF700X_RAW_DATA_RX
+	if (vif_ctx_zep->if_type == NRF_WIFI_IFTYPE_MONITOR) {
```

Comment:
Would it be better to just change this check to `if (vif_ctx_zep->if_type != NRF_WIFI_IFTYPE_STATION)` . If yes, then the conditional (`#ifdef`) can also be removed.

Reply:
as per mode settings - we had different interface types such as STA, STA+TX injector, STA + TXINJECTOR+Promiscuous, STA+promiscuous in UMAC header file. These are not under any CONFIG flag. Let me check if i can allow this for the specific ones. I do not see that STA + TXinjector + Promiscuous and STA + Promiscuous is defined in UMAC file still. Possibly it is missed since Promiscuous changes are still coming in and possibly still not part of new set of changes.

Reply:
keeping this as is at the moment till interface types enums are updated in UMAC header file

Reply:
Are these being defined as `enums` ? Wouldn't a `bit flags` work better here ?

Reply:
This is the definition that is present as an ``enum nrf_wifi_iftype`` in the UMAC header file 
./fw_if/umac_if/inc/fw/host_rpu_umac_if.h. It is not a new enum introduction but rather an extension of the already existing description that UMAC maintained and continues to maintain and refer for its interfaces. This set of interfaces, if the UMAC team have to change and we can borrow the same

Reply:
@rajb9 / @karun2796 ^^^